spec-old-api/02-integration/05-proxy/09-balancer_spec.lua

```diff
@@ -235,234 +237,396 @@ local function http_server(timeout, host, port, counts, test_log)
-    body = body,
-  })
-  if not res then
-    return nil, err
+local add_upstream
+local patch_upstream
+local get_upstream_health
+local add_target
+local add_api
+local patch_api
+local add_plugin
+local delete_plugin
+local gen_port
+do
+  local gen_sym
+  do
+    local sym = 0
+    gen_sym = function(name)
+      sym = sym + 1
+      return name .. "_" .. sym
+    end
+  end
+
+  local function api_send(method, path, body)
+    local api_client = helpers.admin_client()
+    local res, err = api_client:send({
+      method = method,
+      path = path,
+      headers = {
+        ["Content-Type"] = "application/json"
+      },
+      body = body,
+    })
+    if not res then
+      api_client:close()
+      return nil, err
+    end
+    local res_body = res.status ~= 204 and cjson.decode((res:read_body()))
+    api_client:close()
+    return res.status, res_body
+  end
+
+  add_upstream = function(data)
+    local upstream_name = gen_sym("upstream")
+    local req = utils.deep_copy(data) or {}
+    req.name = req.name or upstream_name
+    req.slots = req.slots or SLOTS
+    assert.same(201, api_send("POST", "/upstreams", req))
+    return upstream_name
+  end
+
+  patch_upstream = function(upstream_name, data)
+    assert.same(200, api_send("PATCH", "/upstreams/" .. upstream_name, data))
+  end
+
+  get_upstream_health = function(upstream_name)
+    local path = "/upstreams/" .. upstream_name .."/health"
+    local status, body = api_send("GET", path)
+    assert.same(200, status)
+    return body
+  end
+
+  do
+    local port = FIRST_PORT
+    gen_port = function()
+      port = port + 1
+      return port
+    end
+  end
+
+  add_target = function(upstream_name, host, port, data)
+    port = port or gen_port()
+    local req = utils.deep_copy(data) or {}
+    req.target = req.target or utils.format_host(host, port)
+    req.weight = req.weight or 10
+    local path = "/upstreams/" .. upstream_name .. "/targets"
+    assert.same(201, api_send("POST", path, req))
+    return port
+  end
+
+  add_api = function(upstream_name)
+    local api_name = gen_sym("api")
+    local api_host = gen_sym("host")
+    assert.same(201, api_send("POST", "/apis", {
+      name = api_name,
+      hosts = api_host,
+      upstream_url = "http://" .. upstream_name,
+    }))
+    return api_host, api_name
+  end
+
+  patch_api = function(api_name, new_upstream)
+    assert.same(200, api_send("PATCH", "/apis/" .. api_name, {
+      upstream_url = new_upstream
+    }))
+  end
+
+  add_plugin = function(api_name, body)
+    local path = "/apis/" .. api_name .. "/plugins"
+    local status, plugin = assert(api_send("POST", path, body))
+    assert.same(status, 201)
+    return plugin.id
+  end
+
+  delete_plugin = function(api_name, plugin_id)
+    local path = "/apis/" .. api_name .. "/plugins/" .. plugin_id
+    assert.same(204, api_send("DELETE", path, {}))
   end
-  api_client:close()
-  return res.status
+end
+
+
+local function truncate_relevant_tables()
```

Comment:
this could then receive an argument, `dao` and run the truncations from there instead of the very legacy `helpers.dao` instance